lua/ttt2/libraries/targetid.lua

```diff
@@ -0,0 +1,561 @@
+		walkkey = Key("+walk", "WALK")
+	}
+
+	-- set subtitle depending on the switchmode
+	if switchMode == SWITCHMODE_PICKUP then
+		tData:SetSubtitle(ParT("target_pickup_weapon", key_params_wep) .. (not isActiveWeapon and ParT("target_pickup_weapon_hidden", key_params_wep) or ""))
+	elseif switchMode == SWITCHMODE_SWITCH then
+		tData:SetSubtitle(ParT("target_switch_weapon", key_params_wep) .. (not isActiveWeapon and ParT("target_switch_weapon_hidden", key_params_wep) or ""))
+	elseif switchMode == SWITCHMODE_FULLINV then
+		tData:SetSubtitle(TryT("target_switch_weapon_nospace"))
+	end
+
+	-- add additional dropping info if weapon is switched
+	if switchMode == SWITCHMODE_SWITCH then
+		local dropWepKind = MakeKindValid(dropWeapon.Kind)
+		local dropWeapon_name
+
+		if dropWeapon.GetPrintName then
+			dropWeapon_name = dropWeapon:GetPrintName()
+		end
+
+		dropWeapon_name = dropWeapon_name or dropWeapon.PrintName or dropWeapon:GetClass() or "..."
+
+		tData:AddDescriptionLine(
+			ParT("target_switch_drop_weapon_info", {slot = dropWepKind, name = TryT(dropWeapon_name)}),
+			COLOR_ORANGE
+		)
+	end
+
+	-- add info about full inventory
+	if switchMode == SWITCHMODE_FULLINV then
+		tData:AddDescriptionLine(
+			ParT("target_switch_drop_weapon_info_noslot", {slot = MakeKindValid(ent.Kind)}),
+			COLOR_ORANGE
+		)
+	end
+
+	-- add info if your weapon can not be dropped
+	if switchMode == SWITCHMODE_NOSPACE then
+		tData:AddDescriptionLine(
+			TryT("drop_no_room"),
+			COLOR_ORANGE
+		)
+	end
+end
+
+---
+-- This function handles looking at players and adds specific descriptions
+-- @param TARGET_DATA tData The object to be used in the hook
+-- @realm client
+function targetid.HUDDrawTargetIDPlayers(tData)
+	local client = LocalPlayer()
+	local ent = tData:GetEntity()
+	local obsTgt = client:GetObserverTarget()
+
+	-- has to be a player
+	if not ent:IsPlayer() then return end
+
+	local disguised = ent:GetNWBool("disguised", false)
+
+	-- oof TTT, why so hacky?! Sets last seen player. Dear reader I don't like this as well, but it has to stay that way
+	-- for compatibility reasons. At least it is uncluttered now!
+	client.last_id = disguised and nil or ent
+
+	-- do not show information when observing a player
+	if client:IsSpec() and IsValid(obsTgt) and ent == obsTgt then return end
+
+	-- disguised players are not shown to normal players, except: same team, unknown team or to spectators
+	if disguised and not (client:IsInTeam(ent) and not client:GetSubRoleData().unknownTeam or client:IsSpec()) then return end
+
+	-- show the role of a player if it is known to the client
+	local rstate = GetRoundState()
+	local target_role
+
+	-- TODO: this detective check has to be removed from here and be replaced with a general role flag
+	if ent.GetSubRole and (rstate > ROUND_PREP and ent:IsDetective() or rstate == ROUND_ACTIVE and ent:IsSpecial()) then
+		target_role = ent:GetSubRoleData()
+	end
```

Comment:
So far it seems like this should manage to just get the subroledata of everything that is no inno (IsSpecial) . And apparently simply removing the detective check doesnt change anything I can see, as it still shows the detective icon and no traitor icon, because they are both special.
Anyway, as this is client related I wonder, if the client only knows his friends roles. Because in this code I don't see where this is handled.

So just removing the detective check seems to be sufficient or tell me otherwise :D